spi25.c

```diff
@@ -142,6 +142,11 @@ static int probe_spi_rdid_generic(struct flashctx *flash, int bytes)
 	return 0;
 }
 
+int probe_spi_none(struct flashctx *flash)
+{
+	return 1;
+}
+
```

Comment:
This can't work since it will always succeed when probe_flash() calls it. If the chip can't be probed automatically, then we need to avoid auto-detection all together.

For now, I guess we should set the .probe member in the flashchip struct to NULL and make sure probe_flash() skips NULL entries.

Also, this sort of change belongs in its own patch.